torch/_higher_order_ops/utils.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Callable
+
+import torch
+import torch.utils._pytree as pytree
+
+
+def autograd_not_implemented_inner(
+    operator: Callable, op_name: str, delayed_error: bool, *args: Any, **kwargs: Any
+) -> Any:
+    """If autograd is enabled and any of the arguments require grad this will either
+    raise an error or return a DelayedError depending on the value of delayed.
+
+    Args:
+        operator: The HigherOrderOperator to call with the *args and **kwargs with
+        op_name: The name of the HigherOrderOperator
+        delayed_error: If True, return a DelayedError instead of raising an error
+        args: The flattened operands to the HigherOrderOperator
+        kwargs: The keyword arguments to the HigherOrderOperator
+
+    Raises:
+        RuntimeError: If autograd is enabled and any of the arguments to the HigherOrderOperator
+    """
+    with torch._C._AutoDispatchBelowAutograd():
+        result = operator(*args, **kwargs)
+        flat_operands, _ = pytree.tree_flatten(args)
+        if torch.is_grad_enabled() and any(
+            f.requires_grad for f in flat_operands if isinstance(f, torch.Tensor)
+        ):
+            if delayed_error:
+                err_fn = torch._C._functions.DelayedError(
+                    f"Autograd not implemented for {op_name}",
```

Comment:
Hmm, does the error message look better if we replace op_name with str(operator)?

Reply:
Great call out, you are totally right